### PR TITLE
[4.13 backport] CI jobs for AWS Shared-VPC install

### PR DESCRIPTION
**Pull request by yunjiang29**
New jobs:
* 4.13
  * `aws-ipi-private-shared-vpc-phz-sts-f14 `
  * `aws-ipi-private-shared-vpc-phz-sts-f28-destructive`
  * `aws-ipi-shared-vpc-phz-f14`
  * `aws-ipi-shared-vpc-phz-f28-destructive`
  * `aws-ipi-shared-vpc-phz-sts-fips-f14`
  * `aws-ipi-shared-vpc-phz-sts-fips-f28-destructive`

* 4.13 stable -> 4.14 nightly:
  * `aws-ipi-shared-vpc-phz-sts-fips-f14`
  * `aws-ipi-shared-vpc-phz-f14`

**Note**:  `4.13.0-0.nightly-2023-08-08-191218` is available:
* https://github.com/openshift/api/pull/1529
* https://github.com/openshift/cluster-config-operator/pull/337
* https://github.com/openshift/installer/pull/7362
* https://github.com/openshift/cluster-ingress-operator/pull/966

### CODE REVIEW

**Comment by yunjiang29** [comment]
@gpei @jiajliu @shellyyang1989 PTAL, thanks

**Comment by jiajliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-f14

**Comment by yunjiang29** [comment]
@jiajliu No available build for this feature at this moment as PRs were just merged a few hours ago. I will update the build number in the description once it's ready, sorry for confused info.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-f14

**Comment by yunjiang29** [comment]
Job failed is b/c Prow is still using an old unsupported  version
```
INFO[2023-08-09T05:40:10Z] Resolved release target to registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-08-08-094653 
```

**Comment by yunjiang29** [comment]
@liangxia Can you approve this PR? thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 1]
```diff
@@ -102,7 +102,7 @@ resources:
       memory: 4Gi
 tests:
 - as: alibaba-ipi-private-fips-f28-ui
-  cron: 33 1 29 * *
+  cron: 47 14 13 * *
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:105]:
```
[lxia@homepc tools]$ ./generate-cron-entry.sh alibaba-ipi-private-fips-f28-ui openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml
cron: 33 1 29 * *
```
It looks like the cron entry does not require change on my side.

Comment (gpei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:105]: +1, I think we can keep the other jobs' cron entry as previous, at least as I know the cron time for some of the GCP jobs were reconciled by Jianli on purpose as now.

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:105]: OK, I used `update-cron-entries.py` to do a batch update.
Will fix it now.

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: @liangxia @gpei PATL again, thanks.

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: I also made an update on `update-cron-entries.py` https://github.com/openshift/release/pull/42133 to make sure the output values are consistent, cc @jianlinliu for review

**Pull request merged**